Cubical/HITs/Truncation/Base.agda

```diff
@@ -0,0 +1,13 @@
+{-# OPTIONS --cubical --safe #-}
+module Cubical.HITs.Truncation.Base where
+
+open import Cubical.Data.Nat
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.HLevels
+open import Cubical.HITs.Sn
+
+-- note: only works when n is non-zero
```

Comment:
Maybe we should define the type of nonzero naturals?

Reply:
Or maybe in the definition, instead of `n` I say `suc n`. Therefore `∥A∥0` would actually be prop truncation.

Reply:
That sounds like a better solution than my suggestion! Please do it

Reply:
actually I think it might be easier to keep it this way - otherwise I will be proving `isOfHLevel (suc n) (∥A∥n)` which is kind of awkward and hard to memorize. Much easier to remember `isOfHLevel n (∥A∥n)` for any positive `n`

Reply:
Or maybe even better, define `∥A∥0` to be unit type, and `∥A∥n+1` to be this HIT. Then from user's perspective we have a neat definition.

Reply:
Hmm, the most sensible thing would be to have `||A|| (-1)` be propositional truncation, `||A|| 0` be set truncation, etc. Then you would have `isOfHLevel (suc (suc n)) (||A|| n)` which what one would expect.

Reply:
Well, maybe we should have a `N-2` type from the beginging, then the def. of `isOfHLevel` would also be more sensible.

But anyway, if we do what I said above, we can still have `isOfHLevel n (∥A∥n)` for any natural number `n`

Reply:
The confusing thing is that Vladimir defined the H-levels to start at 0 while the truncation levels in the HoTT book start at -2. So there will always be a off-by-two mismatch between truncations and H-levels. In order to not have any more confusion I think the best is to stick to this convention and have `isOfHLevel (suc (suc n)) ||A||n`.

Reply:
OK. But if we already picked hlevels, doesn't it make sense to stick to it? Use hLevels throughout the project.

Reply:
But ||A||n is n-truncation... Which is of hlevel n+2. I think it would be more confusing to use ||A||n for n+2 truncation than defining ℕ₋₂ as in https://github.com/HoTT/HoTT-Agda/blob/e48a16bcd719e2fcb409d79e3f7df6c6b81223bb/old/Types.agda#L79

Reply:
Starting at 0 or -2 is a completely superficial choice whose only benefit is readability. Given that the HoTT book starts at -2 and very few people use h-levels, I would argue that starting at -2 is the correct choice.